### DISCUSSION TITLE
Is it possible to generate sub-graphs of the schema?

### DISCUSSION THREAD

**Discussion by snendev** [schemaorg/schemaorg#3398] [Q&A]
Hi,

I am very new here so I apologize if this is irrelevant or there is no reasonable solution.

I stumbled upon schema.org while researching existing schemas for an application I am curious to develop. I found the schema definitions to be very thoughtful and useful, so I was interested in extracting a snapshot of the schema for use in my own experiments.

However, I have only found it possible to extract the schema as a whole. Are there options for generating sub-graphs of the schema based on some desired included nodes, or would I have to invoke some parser for the `.ttl` syntax and walk the graph myself? If so, are there existing/recommended tools for doing this, or am I "way out of left field" here? (I can imagine this would be a useful tool, even if maintained elsewhere.)

I am open to suggestion. I can definitely generate my own schema files from the semantic content on the website, but it would be nice to avoid human error and potentially save time.

Thank you for your time! I tried to search for similar discussions but could not find anything related. Perhaps I am using the wrong terms.

**Comment by WeaverStever** [comment]
Not sure what you are asking here.  
The "typed" entities can be referenced through an @id property though.

For instance, in this documentation, the sub graphs are referenced by the @id property.
https://developer.yoast.com/features/schema/functional-specification/

See also:
https://w3c.github.io/json-ld-syntax/#advanced-concepts
